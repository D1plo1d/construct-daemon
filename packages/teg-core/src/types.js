/* config */
export { default as Config, MockConfig } from './config/types/Config'
export * as AxisTypeEnum from './config/types/AxisTypeEnum'
export * as ComponentTypeEnum from './config/types/components/ComponentTypeEnum'

/* jobQueue */
export { default as JobHistoryEvent } from './jobQueue/types/JobHistoryEvent'
export { default as Job, MockJob } from './jobQueue/types/Job'
export { default as JobFile, MockJobFile } from './jobQueue/types/JobFile'
export * as TaskStatusEnum from './jobQueue/types/TaskStatusEnum'
export { default as Task, MockTask } from './jobQueue/types/Task'

/* printer */

/* devices */
export { default as Device } from './devices/types/Device'
export * as DeviceTypeEnum from './devices/types/DeviceTypeEnum'
