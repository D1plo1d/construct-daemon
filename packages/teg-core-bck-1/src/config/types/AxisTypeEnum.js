import { List } from 'immutable'

export const MOVEMENT_AXIS = 'MOVEMENT_AXIS'
export const EXTRUDER_AXIS = 'EXTRUDER_AXIS'

const AxisTypeEnum = List([
  MOVEMENT_AXIS,
  EXTRUDER_AXIS,
])

export default AxisTypeEnum
