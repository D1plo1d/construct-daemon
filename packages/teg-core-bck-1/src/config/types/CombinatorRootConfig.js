import { Record, List } from 'immutable'
import uuid from 'uuid'

import HostConfig from './HostConfig'
import MachineConfig from './MachineConfig'
import AuthConfig from './auth/AuthConfig'
import MaterialConfig from './MaterialConfig'

export const CombinatorRootConfigRecordFactory = Record({
  id: null,
  modelVersion: 0,
  host: null,
  printer: null,
  auth: null,
  materials: List(),
})

const CombinatorRootConfig = ({
  id = uuid.v4(),
  modelVersion = 0,
  host = {},
  machines = [],
  auth = {},
  materials = [],
  ...props
} = {}) => (
  CombinatorRootConfigRecordFactory({
    id,
    modelVersion,
    host: HostConfig(host),
    machines: machines.map(MachineConfig),
    auth: AuthConfig(auth),
    materials: List(materials).map(material => MaterialConfig(material)),
    ...props,
  })
)

export const MockConfig = ({
  host = {},
  printer = {},
  ...props
} = {}) => {
  // eslint-disable-next-line global-require
  const developmentConfig = require('../../../../development.config')
  return CombinatorRootConfig({
    ...developmentConfig,
    host: {
      ...developmentConfig.host,
      ...host,
    },
    machines: [{
      ...developmentConfig.printer,
      ...printer,
    }],
    ...props,
  })
}

export default CombinatorRootConfig
