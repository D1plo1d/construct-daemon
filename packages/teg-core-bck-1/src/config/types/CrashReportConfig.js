import { Record, Map } from 'immutable'
import uuid from 'uuid'

export const CrashReportConfigFactory = Record({
  id: null,
  modelVersion: 0,
  directory: '/var/log/teg',
  uploadCrashReportsToDevs: true,
  model: Map(),
})

const CrashReportConfig = ({
  id = uuid.v4(),
  modelVersion = 0,
  ...props
} = {}) => (
  CrashReportConfigFactory({
    ...props,
    id,
    modelVersion,
    model: Map(props.model),
  })
)

export default CrashReportConfig
