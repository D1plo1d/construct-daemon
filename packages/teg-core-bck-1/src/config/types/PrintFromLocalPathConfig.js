// TODO: Move to a plugin
// import { Record, Map } from 'immutable'
// import t from 'tcomb-validation'
//
// export const PrintFromLocalPathConfigStruct = t.struct({
//   enabled: t.Boolean,
//   allowSymlinks: t.Boolean,
//   whitelist: t.list(t.String),
// })
//
// const PrintFromLocalPathConfig = Record({
//   enabled: false,
//   allowSymlinks: false,
//   whitelist: [
//     '~/.teg-local-files/',
//   ],
// })
// export default PrintFromLocalPathConfig
