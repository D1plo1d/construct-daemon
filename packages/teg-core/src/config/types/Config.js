import { Record, List, Map } from 'immutable'
import uuid from 'uuid'

import HostConfig from './HostConfig'
import PrinterConfig from './PrinterConfig'
import AuthConfig from './auth/AuthConfig'
import MaterialConfig from './MaterialConfig'

export const ConfigRecordFactory = Record({
  id: null,
  modelVersion: 0,
  host: null,
  printer: null,
  machines: Map(),
  auth: null,
  materials: List(),
})

const Config = ({
  id = uuid.v4(),
  modelVersion = 0,
  host = {},
  printer = {},
  auth = {},
  materials = [],
  ...props
} = {}) => {
  const printerConfig = PrinterConfig(printer)

  return ConfigRecordFactory({
    id,
    modelVersion,
    host: HostConfig(host),
    printer: printerConfig,
    machines: Map({ [printerConfig.id]: printerConfig }),
    auth: AuthConfig(auth),
    materials: List(materials).map(MaterialConfig),
    ...props,
  })
}

export const MockConfig = ({
  host = {},
  printer = {},
  ...props
} = {}) => {
  // eslint-disable-next-line global-require
  const developmentConfig = require('../../../../development.config')
  return Config({
    ...developmentConfig,
    host: {
      ...developmentConfig.host,
      ...host,
    },
    printer: {
      ...developmentConfig.printer,
      ...printer,
    },
    ...props,
  })
}

export default Config
