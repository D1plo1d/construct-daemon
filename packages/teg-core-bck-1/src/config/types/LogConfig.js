import { Record, List, Map } from 'immutable'
import uuid from 'uuid'

import { WARNING, ERROR, FATAL } from '../../log/types/logLevelEnum'

export const LogConfigFactory = Record({
  id: null,
  modelVersion: 0,
  maxLength: 1000,
  stderr: List([WARNING, ERROR, FATAL]),
  model: Map(),
})

const LogConfig = ({
  id = uuid.v4(),
  modelVersion = 0,
  model = {},
  ...props
} = {}) => (
  LogConfigFactory({
    ...props,
    id,
    modelVersion,
    model: Map(model),
  })
)

export default LogConfig
