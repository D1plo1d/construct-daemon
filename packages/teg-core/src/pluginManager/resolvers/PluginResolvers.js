import { NullSchemaForm } from '../types/SchemaForm'

const PluginResolvers = {
  Plugin: {
    configForm: (source, args, { store }) => {
      const { id, model, modelVersion } = source

      const state = store.getState()
      const schemaForm = state.schemaForms.getIn(
        ['plugins', source.package],
        NullSchemaForm,
      )

      return {
        id,
        model,
        modelVersion,
        schemaForm,
      }
    },
  },
}

export default PluginResolvers
