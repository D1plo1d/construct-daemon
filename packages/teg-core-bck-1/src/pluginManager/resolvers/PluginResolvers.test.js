import { Record, Map } from 'immutable'
import typeDefs from '@tegapp/schema'

import deterministicTestSetup from '../../util/testing/deterministicTestSetup'
import snapshotTestResolvers from '../../util/testing/snapshotTestResolvers'

import { MockConfig } from '../../config/types/Config'
import SchemaForm from '../types/SchemaForm'

import PluginResolvers from './PluginResolvers'

describe('PluginResolvers', () => {
  it('presents a consistent API', () => {
    deterministicTestSetup()

    const state = Record({
      schemaForms: Record({
        plugins: Map({
          '@tegapp/core': SchemaForm({
            id: '@tegapp/core',
          }),
        }),
      })(),
    })()

    snapshotTestResolvers({
      typeDefs,
      typeName: 'Plugin',
      resolvers: PluginResolvers,
      rootValue: MockConfig().printer.plugins.find(p => (
        p.package === '@tegapp/core'
      )),
      contextValue: {
        store: {
          getState: () => state,
        },
      },
    })
  })
})
