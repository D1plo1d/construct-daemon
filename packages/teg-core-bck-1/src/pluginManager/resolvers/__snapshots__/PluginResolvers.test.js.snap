// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PluginResolvers presents a consistent API 1`] = `
Object {
  "_testPlugin": Object {
    "configForm": Object {
      "id": "asgbvas23",
    },
    "id": "asgbvas23",
    "package": "@tegapp/core",
  },
}
`;
